app/validators/note_length_validator.rb

```diff
@@ -8,7 +8,7 @@ def validate_each(record, attribute, value)
   private
 
   def too_long?(value)
-    countable_text(value).mb_chars.grapheme_length > options[:maximum]
+    countable_text(value).length > options[:maximum]
```

Comment:
This is not equivalent.

```
mastodon(test)> '💖✨🏳️‍⚧️'.length
=> 7
mastodon(test)> '💖✨🏳️‍⚧️'.mb_chars.grapheme_length
=> 3
```

I think the equivalent would be `countable_text(value).grapheme_clusters.size`, or maybe `countable_text(value).each_grapheme_cluster.size` to avoid allocating an array (seems to be a bit faster than the former one or the current implementation in my limited benchmark).

Reply:
Ah ... the change handles multi-byte ok, but not ZWJ style emoji. Updated to add some coverage for these various length cases, and to use `each_grapheme_cluster` in the relevant spots.

Possible future refactor here -- to the extent we have some repeated logic around counting the characters (and maybe the URL length special treatment?) repeated in a few spots (at least these three validators, maybe more?) - extract that to a class.